src/linear_algebra/quadratic_form.lean

```diff
@@ -388,6 +389,21 @@ quadratic_form.ext $ λ x,
 
 end associated
 
+section nondegenerate
+
+/-- A non-degenerate quadratic form is zero only on zero vectors. -/
+def nondegenerate (Q : quadratic_form R M) : Prop := ∀ x, Q x = 0 → x = 0
```

Comment:
Is this definitely the standard terminology?

Ignoring characteristic 2 for simplicity, I would refer to a quadratic form as non-degenerate when its associated bilinear form is non-degenerate.

E.g., I would say `Q(x, y) = x² - y²` is non-degenerate (in two dimensions) even though `Q(1, 1) = 0`.

Reply:
(just found the Zulip thread, reposting there)

Reply:
Sorry, I hit the merge button too quickly. We should fix this.